libraries/resource_provider.rb

```diff
@@ -186,6 +186,7 @@ def create(method = :create)
     # Adds Oneview Scope to the Oneview resource if scope is not already added
     def add_to_scopes
       raise "ResourceNotFound: #{@resource_name} '#{@name}' does not exist" unless @item.retrieve!
+      raise "InvalidParameters: Parameters 'operation' must be valid for add_to_scopes" if @new_resource.scopes.nil? || @new_resource.scopes.empty?
```

Comment:
Just skip instead raising an exception, in an automatic gathering recipe it would fail unintendedly